components/clarinet-sdk/vitest-helpers/src/clarityValuesMatchers.ts

```diff
@@ -0,0 +1,377 @@
+    return simpleAssertion.call(this, ClarityType.StringUTF8, actual, Cl.stringUtf8(expected));
+  },
+
+  toBeOk(actual: unknown, expectedValue: ExpectStatic | ClarityValue) {
+    return simpleCompositeAssertion.call(this, ClarityType.ResponseOk, actual, expectedValue);
+  },
+
+  toBeErr(actual: unknown, expectedValue: ExpectStatic | ClarityValue) {
+    return simpleCompositeAssertion.call(this, ClarityType.ResponseErr, actual, expectedValue);
+  },
+
+  toBeSome(actual: unknown, expectedValue: ExpectStatic | ClarityValue) {
+    return simpleCompositeAssertion.call(this, ClarityType.OptionalSome, actual, expectedValue);
+  },
+
+  toBeNone(actual: unknown) {
+    const expectedType = ClarityType.OptionalNone;
+    try {
+      const isCV = checkCVType(actual, expectedType, this.isNot);
+      assert(isCV);
+    } catch (e: any) {
+      return errorToAssertionResult.call(this, e);
+    }
+
+    const expected = Cl.none();
+    return {
+      pass: this.equals(actual, expected, undefined, true),
+      message: () => formatMessage.call(this, formatCV(actual), formatCV(actual)),
+      actual: formatCV(actual, 2),
+      expected: formatCV(actual, 2),
+    };
+  },
+
+  toBePrincipal(actual: unknown, expectedString: string) {
+    const isStandard = !expectedString.includes(".");
+    let expected: StandardPrincipalCV | ContractPrincipalCV;
+
+    const expectedType = isStandard ? ClarityType.PrincipalStandard : ClarityType.PrincipalContract;
+    try {
+      const isCV = checkCVType(actual, expectedType, this.isNot);
+      assert(isCV);
+    } catch (e: any) {
+      return errorToAssertionResult.call(this, e);
+    }
+
+    const actualString = principalToString(actual);
+
+    try {
+      expected = isStandard
+        ? Cl.standardPrincipal(expectedString)
+        : Cl.contractPrincipal(...(expectedString.split(".") as [string, string]));
+    } catch {
+      return {
+        pass: false,
+        message: () => `expected ${expectedString} ${notStr(this.isNot)}to be a principal`,
+        actual: actualString,
+        expected: expectedString,
+      };
+    }
+
+    return {
+      pass: this.equals(actual, expected, undefined, true),
+      message: () => formatMessage.call(this, actualString, expectedString),
+      actual: formatCV(actual, 2),
+      expected: formatCV(expected, 2),
+    };
+  },
+
+  toBeBuff(actual: unknown, expectedRaw: Uint8Array) {
+    const expectedType = ClarityType.Buffer;
+    try {
+      const isCV = checkCVType(actual, expectedType, this.isNot);
+      assert(isCV);
+    } catch (e: any) {
+      return errorToAssertionResult.call(this, e);
+    }
+
+    const expected = Cl.buffer(expectedRaw);
+    return {
+      pass: this.equals(actual, expected, undefined, true),
+      // note: throw a simple message and rely on `actual` and `expected` to display the diff
+      message: () => `the received Buffer does ${this.isNot ? "" : "not "}match the expected one`,
```

Comment:
isNot switched? Could use the `notStr` helper. (and some following cases)

Reply:
Maybe I'm reading this wrong, but the notStr is the opposite, and the tests cases seem to cover these, so idk

Reply:
Yes because the notStr helpers is used for this kind of sentence:

```
expected "x" to be "y"
```

Which could also be
```
the received "y" is not equal to "x"
```

So the "not" isn't used in the same way.

Maybe I should rephrase
```
the received Buffer does not match the expected one
into:
expected the received Buffer to match the expected one"
```

It would be more standard

Reply:
Ah right, makes sense. This is not the "must" style here. 👍🏻